go/libkb/config.go

```diff
@@ -410,7 +410,7 @@ func (f JSONConfigFile) GetAllUsernames() (current NormalizedUsername, others []
 			return
 		}
 		nu := NewNormalizedUsername(name)
-		if !nu.Eq(current) {
+		if !nu.Eq(current) && !nu.IsNil() {
```

Comment:
Instead of doing this, it's better to use `GetAllProvisionedUsernames` instead.  That's going to be more correct for various scenarios, such as when the device is question was offline at the time of the reset.

Reply:
And BTW, the bug with the code that's currently there is that the `currentUsername` is coming back as empty, since the current user doesn't exist after the reset.

Reply:
sorry, Max, I don't quite follow. `GetAllProvisionedUsernames` instead of what, exectly? `SecretStoreContext.GetAllUserNames` ? And how to resolve the right context to call `GetAllProvisionedUsernames` from inside the secret store?

Reply:
See the comment below. Instead of `currentUsername, otherUsernames, err := c.GetAllUserNames()`, call `GetAllProvisionedUsernames`.  Change `GetConfiguredAccounts` to take a `MetaContext` and pass that through.

Reply:
Can you remove this `nu.IsNil` check here?  I don't think it's needed.